test/Serilog.Tests/Formatting/Json/JsonFormatterTests.cs

```diff
@@ -225,5 +225,51 @@ public void SequencesOfSequencesAreSerialized()
             var h = (string)d.Properties.AProperty[0][0];
             Assert.Equal("Hello", h);
         }
+
+
+        [Fact]
+        public void ABadBehaviorClassAsSerialized()
+        {
+            var p = new MessageTemplateParser();
+            var e = new LogEvent(Some.OffsetInstant(), LogEventLevel.Information, null,
+                p.Parse("{@ABadBehavior}"), new[] { new LogEventProperty("ABadBehavior", new ScalarValue(new ABadBehavior())) });
+
+            var d = FormatEvent(e);
+
+            var h = (string)d.Properties.ABadBehavior;
+            Assert.Null(h);
+        }
+
+        [Fact]
+        public void ABadBehaviorClassAsStringification()
+        {
+            var p = new MessageTemplateParser();
+            var e = new LogEvent(Some.OffsetInstant(), LogEventLevel.Information, null,
+                p.Parse("{$ABadBehavior}"), new[] { new LogEventProperty("ABadBehavior", new ScalarValue(new ABadBehavior())) });
+
+            var d = FormatEvent(e);
+
+            var h = (string)d.Properties.ABadBehavior;
+            Assert.Null(h);
+        }
+
+        [Fact]
+        public void ABadBehaviorClassAsRenderings()
+        {
+            var p = new MessageTemplateParser();
+            var e = new LogEvent(Some.OffsetInstant(), LogEventLevel.Information, null,
+                p.Parse("{ABadBehavior}"), new[] { new LogEventProperty("ABadBehavior", new ScalarValue(new ABadBehavior())) });
```

Comment:
Now I think I created this UnitTes wrongly. The ScalarValue can never have an instance of an object like this in the normal execution of the Serilog pipeline.